packages/utils/plugin/widget.ts

```
@@ -90,3 +131,16 @@ export function isAllowedWidgetModule(moduleName: string): boolean {
 }
+
+export interface TouchWidgetDefinition<TProps = Record<string, unknown>> {
+  runtime?: WidgetRuntime
+  render?: unknown
+  setup?: (props: TProps) => unknown
+  tagName?: string
+  define?: () => void
+  [key: string]: unknown
+}
+
+export function defineTouchWidget<T extends TouchWidgetDefinition | Function>(widget: T): T {
```

ESLint

[error] The `Function` type accepts any function-like value.
Prefer explicitly defining any function parameters and return type.

(ts/no-unsafe-function-type)